mobile-app/app/screens/AppNavigator/screens/Balances/components/AddressControlScreen.tsx

```diff
@@ -0,0 +1,192 @@
+import { BalanceParamList } from '../BalancesNavigator'
+import { useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { RandomAvatar } from '@screens/AppNavigator/screens/Balances/components/RandomAvatar'
+
+export function AddressControlScreen (): JSX.Element {
+  const navigation = useNavigation<NavigationProp<BalanceParamList>>()
+  return (
+    <ThemedScrollView>
+      <ThemedSectionTitle
+        testID='switch_address_screen_title'
+        text={translate('screens/OnboardingNetworkSelectScreen', 'Switch to another wallet')}
+      />
+      <AddressControlCard onClose={() => navigation.goBack()} />
+    </ThemedScrollView>
+  )
+}
+
+export function AddressControlModal ({ onClose }: { onClose: () => void }): JSX.Element {
+  return (
+    <View style={tailwind('w-full pb-16')}>
+      <ThemedView
+        dark={tailwind('border-gray-700')}
+        light={tailwind('border-gray-100')}
+        style={tailwind('border-b-2')}
+      >
+        <View style={tailwind('flex flex-row justify-between w-full px-4 pb-4 pt-2')}>
+          <ThemedText
+            dark={tailwind('text-gray-50')}
+            light={tailwind('text-gray-900')}
+            style={tailwind('ml-2 text-lg font-medium')}
+          >
+            {translate('screens/AddressControlScreen', 'Switch to another wallet')}
+          </ThemedText>
+          <TouchableOpacity onPress={onClose}>
+            <ThemedIcon
+              size={24}
+              name='close'
+              iconType='MaterialIcons'
+              dark={tailwind('text-white text-opacity-70')}
+              light={tailwind('text-gray-600')}
+            />
+          </TouchableOpacity>
+        </View>
+      </ThemedView>
+      <ThemedScrollView
+        dark={tailwind('text-gray-50')}
+        light={tailwind('text-gray-900')}
+        contentContainerStyle={tailwind('pb-8')}
+      >
+        <AddressControlCard onClose={onClose} />
+      </ThemedScrollView>
+    </View>
+  )
+}
+
+export function AddressControlCard ({ onClose }: { onClose: () => void }): JSX.Element {
+  const { address, addressLength, setIndex, wallet } = useWalletContext()
+  const [availableAddresses, setAvailableAddresses] = useState<string[]>([])
+  const [canCreateAddress, setCanCreateAddress] = useState<boolean>(false)
+  const logger = useLogger()
+
+  const discoverAddresses = async (): Promise<void> => {
+    const addresses: string[] = []
+    for (let i = 0; i <= addressLength; i++) {
+      const account = wallet.get(i)
+      const address = await account.getAddress()
+      addresses.push(address)
+    }
+    setAvailableAddresses(addresses)
+    const discoveredAddress = await wallet.discover(addresses.length)
+    setCanCreateAddress(addresses.length === discoveredAddress.length)
+  }
+
+  const onRowPress = async (index: number): Promise<void> => {
+    await setIndex(index)
+    onClose()
+  }
+
+  useEffect(() => {
+    discoverAddresses().catch(logger.error)
```

Comment:
Should set loading indicator for this since it's an expensive and long operation.

Reply:
After adding confirm pin, we already have loader saying `It may take a few seconds to securely encrypt your wallet...` this loader will be there until we don't discover all active address.